src/node_modules/_file-explorer.js

```diff
@@ -0,0 +1,292 @@
+var setval = require('setval')
+var yo = require('yo-yo')
+var csjs = require('csjs-inject')
+var Treeview = require('ethereum-remix').ui.TreeView
+
+var css = csjs`
+  .treeview {
+    display: block;
+    margin-top: 7px;
+    width: 20%;
+  }
+  .folder,
+  .file {
+    font-size: 14px;
+  }
+  .hasFocus {
+    background-color: #F4F6FF;
+  }
+  .rename {
+    background-color: white;
+  }
+  .remove,
+  .save {
+    align-self: center;
+    margin-right: 3%;
+  }
+  .activeMode {
+    display: flex;
+    justify-content: space-between;
+  }
+`
+module.exports = fileExplorer
+
+function fileExplorer (container, files, editor) {
```

Comment:
I agree that `app.js` should remain as simple as possible. The idea would be that `app.js` is just there to connect the endpoints between the modules. Please also keep in mind that it should be possible to use remix (and potentially other modules) in a different way than they are used currently. For example, if the whole system would be used together with an existing IDE, the existing IDE would provide the editor and we just have to connect the modules differently.